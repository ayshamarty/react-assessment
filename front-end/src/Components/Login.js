import React, { Component } from "react";
import axios from "axios";

class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      thisUser: [],
      users: this.props.userdata,
      errors: "",
      passMatch: ""
    };
  }

  getOne = (userToGet, passwordToGet) => {
    this.setState({
      errors: "",
      passMatch: ""
    });
    axios
      .get(`http://localhost:5000/user/name/${userToGet}`)
      .then(response => {
        this.setState({
          thisUser: response.data
        });

        let thisUser = this.state.thisUser;
        let notifier = document.getElementById("notifier");
        // console.log(thisUser[0]);

        thisUser.forEach(user => {
          if (user.password === passwordToGet) {
            this.setState({
              passMatch: "match"
            });
            return "";
          } else {
            this.setState({
              errors: "no match"
            });
          }
        });

        if (this.state.passMatch === "") {
          notifier.style.color = "red";
          notifier.innerText = "incorrect username and password combination";
        } else {
          notifier.style.color = "black";
          notifier.innerText = "logged in";
        }
      })
      .catch(err => {
        let notifier = document.getElementById("notifier");
        notifier.style.color = "red";
        notifier.innerText = "please enter a valid username";
      });
  };

  loginUser = e => {
    e.preventDefault();
    const userlogin = {
      username: e.target[0].value,
      password: e.target[1].value
    };

    this.getOne(userlogin.username, userlogin.password);
  };
  render() {
    return (
      <div className="container">
        <form onSubmit={this.loginUser}>
          <h3>Please enter your details to login</h3>
          <br />
          <p>username</p>
          <input id="input-username" />
          <br />
          <p>password</p>
          <input type="password" id="input-password" />
          <button type="submit">Login</button>
        </form>
        <div>
          <span id="notifier" />
        </div>
      </div>
    );
  }
}

export default Login;
