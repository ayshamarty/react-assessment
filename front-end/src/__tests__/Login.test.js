import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import Login from "../Components/Login";

import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Login />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders shallow without crashing", () => {
  shallow(<Login />);
});

it('snapshot Login', () => {
  const wrapper = renderer.create(<Login/>).toJSON()
  expect(wrapper).toMatchSnapshot();
})


