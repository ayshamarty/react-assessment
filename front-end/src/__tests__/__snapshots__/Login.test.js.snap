// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Login 1`] = `
<div
  className="container"
>
  <form
    onSubmit={[Function]}
  >
    <h3>
      Please enter your details to login
    </h3>
    <br />
    <p>
      username
    </p>
    <input
      id="input-username"
    />
    <br />
    <p>
      password
    </p>
    <input
      id="input-password"
      type="password"
    />
    <button
      type="submit"
    >
      Login
    </button>
  </form>
  <div>
    <span
      id="notifier"
    />
  </div>
</div>
`;
